detd/service.py

```diff
@@ -249,14 +248,19 @@ def _add_talker(self, request):
         interval = request.period
         size = request.size
         interface_name = request.interface
+        tx_selection = request.tx_selection
+        tx_selection_offload = request.tx_selection_offload
+        data_path = request.data_path
+        preemption = request.preemption
+        launch_time_control = request.launch_time_control
+
 
-        options = Options()
-        options.qdiscmap = request.qdiscmap
         interface = Interface(interface_name)
         stream = StreamConfiguration(addr, vid, pcp, txoffset)
         traffic = TrafficSpecification(interval, size)
+        hints = Hints(TxSelection, False, DataPath, True, True)
```

Comment:
The parameters seem to be a typo? I guess the intent is to refer to the variables that you define in lines 251 to 255, and initialize from the request content?